igvc_navigation/src/path_follower/main.cpp

```diff
@@ -48,113 +64,78 @@ void position_callback(const nav_msgs::OdometryConstPtr& msg)
     return;
   }
 
+  double end_x = waypoint->point.x;
```

Comment:
I would in line these with the function call for clarity, i.e. use waypoint->point.x when you get the distance